src/celeritas/neutron/interactor/detail/InvariantQ2Sampler.hh

```diff
@@ -0,0 +1,363 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/interactor/detail/InvariantQ2Sampler.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <cmath>
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/random/distribution/BernoulliDistribution.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Sample the momentum transfer (\f$ Q^{2} = -t \f$) of the neutron-nucleus
+ * elastic scattering.
+ *
+ * \note This performs the same sampling routine as in Geant4's
+ *  G4ChipsElasticModel and G4ChipsNeutronElasticXS where the differential
+ *  cross section for quark-exchange and the amplitude of the scattering are
+ *  parameterized in terms of the neutron momentum (GeV/c) in the lab frame.
+ */
+class InvariantQ2Sampler
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Mass = units::MevMass;
+    //!@}
+
+  public:
+    // Construct with shared and target data, and the neutron momentum
+    inline CELER_FUNCTION InvariantQ2Sampler(NeutronElasticRef const& shared,
+                                             IsotopeView const& target,
+                                             real_type neutron_p);
+
+    // Sample the momentum transfer
+    template<class Engine>
+    inline CELER_FUNCTION real_type operator()(Engine& rng);
```

Comment:
Is the result `MevMomentum`? It should definitely be a `Quantity` then. I think you can also use quantity to handle the transfer to and from GeV internally...

Reply:
Actually the result is $({\rm MevMomentum})^{2}$.